client/state/data-layer/wpcom/reader/data-request/index.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Internal dependencies
+ */
+import { READER_DATA_REQUEST } from 'state/action-types';
+
+export const readerDataHandler = ( { dispatch }, {
+	dataFetch,
+	dispatchArgs = {},
+	requestAction,
+} ) => {
+
+	dispatch( {
+		type: requestAction,
+		...dispatchArgs,
+	} );
+
+	return dataFetch()
+		.then(
+			data => {
+				dispatch( {
+					type: `${ requestAction }_SUCCESS`,
+					...data,
+					...dispatchArgs,
+				} );
+			} )
+		.catch(
+			error => {
+				dispatch( {
+					type: `${ requestAction }_FAILURE`,
+					error,
+					...dispatchArgs,
+				} );
+			}
+		);
+};
+
+export default {
+	[ READER_DATA_REQUEST ]: [ readerDataHandler ],
+};
```

Comment:
Is there a `/wpcom/rest/v1/reader/data-request/` API endpoint?

It would be good to keep these files structured to mirror what exists on WordPress.com
https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer#file-structure